server/src/main/java/org/apache/cassandra/sidecar/restore/RestoreJobDiscoverer.java

```diff
@@ -385,6 +399,41 @@ private void processSidecarManagedJobMaybe(RestoreJob job)
         }
     }
 
+    private void initLocalDatacenterMaybe()
+    {
+        if (localDatacenter != null)
+        {
+            return;
+        }
+
+        try
+        {
+            NodeSettings nodeSettings = instanceMetadataFetcher.callOnFirstAvailableInstance(i -> i.delegate().nodeSettings());
+            localDatacenter = nodeSettings.datacenter();
+        }
+        catch (CassandraUnavailableException cue)
+        {
+            LOGGER.debug("localDatacenter is not initialized", cue);
+        }
+    }
+
+    private boolean isLocalDatacenterExcluded(RestoreJob job)
+    {
+        if (!job.shouldRestoreToLocalDatacenterOnly)
+        {
+            return false;
+        }
+
+        if (localDatacenter == null)
+        {
+            LOGGER.debug("The restore job should restore only to the local datacenter, but the local datacetner is undetermined yet; skip this run");
+            return true;
+        }
+
+        // when job should restore to local datacenter only, but the target datacenter is not the local one
+        return !localDatacenter.equalsIgnoreCase(job.localDatacenter);
```

Comment:
datacenter is case sensitive, so I don't think we should equal here with ignore case, even though it would be bad if someone decided to have `DATACENTER1` and `datacenter1`, but it would be possible